### PR TITLE
Fix #2818, replacing Zend_Date with native PHP DateTime object/functions

### LINKED ISSUE #2818: Magento 2.4.6 compatibility
Magento 2.4.6-beta1 is available for partners.

Let's make Elasticsuite compatible with it !

https://experienceleague.adobe.com/docs/commerce-operations/release/notes/adobe-commerce/2-4-6.html

https://developer-stage.adobe.com/commerce/php/development/backward-incompatible-changes/highlights/#246-beta

Notable changes : 

- Removal of "Zend_*" libraries
- Support of Elasticsearch 8 (and OpenSearch 2)

High probability to release a 2.11 branch considering all those backward incompatible changes.

### PR DESCRIPTION
**Pull request by vahonc**

### CODE REVIEW

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 1]
```diff
@@ -31,6 +37,21 @@ class IndexSettings extends AbstractConfiguration
...
 
+    /**
+     * @var string
+     */
+    const INDICES_SETTINGS_CONFIG_XML_INDICES_PATTERN_PATH = 'smile_elasticsuite_core_base_settings/indices_settings/indices_pattern';
+
+    /**
+     * @var string
+     */
+    const DEFAULT_INDICES_PATTERN_NAME_OLD = '{{YYYYMMdd}}_{{HHmmss}}';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Helper/IndexSettings.php:48]: Rename this one to "OLD_DEFAULT_INDICES_PATTERN"

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 2]
```diff
@@ -31,6 +37,21 @@ class IndexSettings extends AbstractConfiguration
...
+
+    /**
+     * @var string
+     */
+    const DEFAULT_INDICES_PATTERN_NAME_NEW = '{{Ymd}}_{{His}}';
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Helper/IndexSettings.php:53]: Rename this one to "DEFAULT_INDICES_PATTERN"

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 3]
```diff
@@ -93,41 +143,47 @@ public function getLanguageCode($store)
...
 
     /**
      * Get index name suffix.
      *
-     * @param Zend_Date $date Date
+     * @param DateTime $date Date
      * @return string
      */
-    public function getIndexNameSuffix(Zend_Date $date): string
+    public function getIndexNameSuffix(DateTime $date): string
```
Comment (romainruaud) [inline src/module-elasticsuite-core/Helper/IndexSettings.php:166]: Can you try to add a unit test to this method as well ? 

The test should mock the fact that the pattern can be the old one or the new one.

File: src/module-elasticsuite-core/Helper/IndexSettings.php (hunk 1 above)
Comment (vahonc) [reply src/module-elasticsuite-core/Helper/IndexSettings.php]: Done.

File: src/module-elasticsuite-core/Helper/IndexSettings.php (hunk 2 above)
Comment (vahonc) [reply src/module-elasticsuite-core/Helper/IndexSettings.php]: Done.

File: src/module-elasticsuite-core/Helper/IndexSettings.php [hunk 4]
```diff
@@ -93,41 +143,47 @@ public function getLanguageCode($store)
...
     {
         /*
         * Generate the suffix of the index name from the current date.
-        * e.g : Default pattern "{{YYYYMMdd}}_{{HHmmss}}" is converted to "20160221_123421".
+        * e.g : Default pattern "{{Ymd}}_{{His}}" is converted to "20160221_123421".
         */
         $indexNameSuffix = $this->getIndicesSettingsConfigParam('indices_pattern');
 
+        if ($indexNameSuffix === self::DEFAULT_INDICES_PATTERN_NAME_OLD) {
```
[the remark this answers is not in the document]
Comment (vahonc) [reply src/module-elasticsuite-core/Helper/IndexSettings.php]: Done.

**Pull request merged**